docs/webui_overriding.md

```diff
@@ -56,6 +56,31 @@ CSS file or a Polymer CSS file (you need to make sure `type` is correct in the
 }
 ```
 
+## Lit Functions
+
+Adding a new function to a Lit component can be complicated because the types
+for the HTML template depend on what's exported, so naively super classing
+the upstream component does not work. The best approach is to use
+[declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)
+and modify the prototype of the upstream class.
+
+```ts
+import { FancyElement } from './fancy-chromium.js'
+
+declare module './item-chromium.js' {
+  interface FancyElement {
+    isBraveAndFancy(): boolean
+  }
+}
+
+ExtensionsItemElement.isBraveAndFancy = () => true;
```

Comment:
Should it be `ExtensionsItemElement.prototype.isBraveAndFancy` ?

Reply:
:facepalm: yes. yes it should.
https://github.com/brave/brave-core/pull/29616